### ISSUE TITLE
This driver might cause data-loss

### ISSUE THREAD

**Issue by systemofapwne** [systemofapwne/mousedriver#3]
TL;DR: **Do not use this driver for the moment, since it might cause data corruption on your storage!** I have to investigate this.

**Long story**
When using this driver while working on it, I suffered from BTRFS data corruption, which spammed the kernel log (I generally have it open for debugging this driver).
Moving away from the WD240 SSD (known to cause at least similar problems on FreeBSD systems regarding TRIM) to a Samsung 870 Evo with a freshly installed Manjaro system did not help. I quickly found [this entry on the kernel bugtracker](https://bugzilla.kernel.org/show_bug.cgi?id=201693), which seemed to be reasonable: I had the SSD attached to a problematic storage controler (AS Media) and people reported issues with that specific drive in combination with that controler.
I was reproduceably able to trigger BTRFS corruption everytime I cloned a 12 GB git repo to my SSD. Running **btrfs scrub** revealed some of freshly written files to be corrupted.
However, after systematically trying several suggested configurations (no overclocking, disabling NCQ, latest firmware/BIOS, different power supply for the SSD), I was suddenly unable to reproduce the issue.
And everytime I did not see an issue was, when I did not load this kernel module.

I am now investigating to reproduce the issue to pinpoint it further down. 
I highly suspect one of the following two things to cause the issue.
1) Using floating-point arithmetic within the kernel (even with the kernel_fpu_begin()/kernel_fpu_end())
2) My latest addition for querying the kernel for the polling rate (within the kernel_fpu_begin()/kernel_fpu_end() block).

If 1) is the reason, then every branch is affected. If it is 2), only the testing branch is affected.

I will investigate this shortly.

**Comment by systemofapwne** [comment]
I was able to confirm issues even with the master branch.

So I repeat: **DO NOT USE THIS KERNELMODULE RIGHT NOW** until I have found a solution

**Comment by systemofapwne** [comment]
I think I have found the root issue. Using the compile flag -msse for a kernel module seems to cause unforseen consequences.
I switched to -mhard-float instead.

SSE anyway only was added to return a float value for a function in the original fork for the fast inverse square-root (Q_sqrt). I switched that over to pass-by-pointer, so the code compiles without the need of SSE.

Oh man, this years old kernel module I forked seem to have been very VERY unsafe.

**Comment by systemofapwne** [comment]
Even though that potential data-loss (especially on BTRFS and RAID6 systems) is now fixed in the master branch, it does not feature a "workaround" with mhard-float, that is already implemented in testing.
Without that workaround, the pointer might jump to a specific corner of your screen and gets stuck there. I was able to trigger this when playing this video in brace/chrome/chromium https://sps-tutorial.com/was-ist-eine-sps/

The ultimate fix would be, to go back to msse. However, even though this compile-flag now is ultimately save, I will probably stay at mhard-float for the moment and decide on this later, once I merge master with testing.
